### PR TITLE
DOC: #38067 add missing holiday observance rules

### LINKED ISSUE #38067: DOC: Include missing holiday observance rules
#### Location of the documentation

[Timeseries Holiday](https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#holidays-holiday-calendars) section in doc/source/user_guide/timeseries.rst.

#### Documentation problem

The list of available holiday observance rules is incomplete - several available observances are missing.

#### Suggested fix for documentation

The following observances should be added to the documentation because they are already available:

- next_workday
- previous_workday
- before_nearest_workday
- after_nearest_workday

Before I examined `holiday` via TAB-completion I had written my own version of weekend_to_monday; others may likewise re-invent these wheels if they're not documented.

(Updated to remove "weekend_to_monday" because it does the same thing as "next_monday")

### PR DESCRIPTION
**Pull request by dontgoto**
- [x] closes #38067

Updated docs with missing functions.

`weekend_to_monday` and `next_monday` do indeed do the exact same thing, and I agree that `next_monday` should be deprecated. Any thoughts?

### CODE REVIEW

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @dontgoto 

> weekend_to_monday and next_monday do indeed do the exact same thing, and I agree that next_monday should be deprecated. Any thoughts?

Yeah that makes sense to deprecate one of them if they are redundant

**Comment by dontgoto** [comment]
> Yeah that makes sense to deprecate one of them if they are redundant

Can I put in a deprecation notice for 4.0.0? Or something like that?

**Comment by Jeitan** [comment]
My two cents here - I think "weekend_to_monday" is more accurate and descriptive, BUT if that's the one we keep it would make the most sense to also change the current "previous_friday" to "weekend_to_friday".  I think it's clearest to keep the before/after and next/previous pairs as matched as possible ... so whichever gets deprecated, the pair should be either:

* weekend_to_monday and weekend_to_friday
* next_monday and previous_friday

I prefer the former but wouldn't care too much if folks prefer the latter.

File: doc/source/user_guide/timeseries.rst [hunk 1]
```diff
@@ -1468,11 +1468,16 @@ or some other non-observed day.  Defined observance rules are:
     :header: "Rule", "Description"
     :widths: 15, 70
 
+    "next_workday", "move Saturday and Sunday to Monday"
+    "previous_workday", "move Saturday and Sunday to Friday"
```
Comment (Jeitan) [inline doc/source/user_guide/timeseries.rst:1472]: This one isn't *quite* right ... it moves to the previous workday even if the input is a weekday, not just for Sat and Sun (as opposed to what previous_friday does)

**Comment by Jeitan** [comment]
@dontgoto I agree on the "makes more sense" ... even though the others have been documented I think that can be got around by deprecating them for a while before they go away

**Pull request merged**